CONTRIBUTING.md

```diff
@@ -188,11 +188,58 @@ with one another are rolled up.
 Speaking of tests, Rust has a comprehensive test suite. More information about
 it can be found [here][rctd].
 
-### External Dependencies
+### External Dependencies (subrepo)
 
-Currently building Rust will also build the following external projects:
+As a developer to this repository, you don't have to treat the following external projects
+differently from other crates that are directly in this repo:
 
 * [clippy](https://github.com/rust-lang/rust-clippy)
+
+They are just regular files and directories. This is in contrast to `submodule` dependencies.
+
+If you want to synchronize or otherwise work with subrepos, install the `git subrepo` command via
+instructions found at https://github.com/ingydotnet/git-subrepo
+
+#### Creating a new subrepo dependency
```

Comment:
Contributors should rarely have to do this, correct? If so, then it should be mentioned last, and it should be said that this is only for *adding new tools to rustc*.